panel/pane/base.py

```diff
@@ -232,21 +237,24 @@ def _update(self, ref=None, model=None):
     #----------------------------------------------------------------
 
     @classmethod
-    def applies(cls, obj):
+    def applies(cls, obj: Any) -> Optional[Union[float, bool]]:
         """
-        Given the object return a boolean indicating whether the Pane
-        can render the object. If the priority of the pane is set to
-        None, this method may also be used to define a priority
+        Returns boolean or float indicating whether the Pane
+        can render the object.
+        
+        If the priority of the pane is set to
+        `None`, this method may also be used to define a float priority
         depending on the object being rendered.
         """
         return None
 
-    def clone(self, object=None, **params):
+    def clone(self: T, object: Optional[Any]=None, **params) -> T:
```

Comment:
The should be the way to type annotate a factory method. See https://github.com/python/typing/issues/58#issuecomment-326240794